hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/RingbufferService.java

```diff
@@ -164,33 +165,30 @@ public RingbufferContainer getContainer(String name) {
             return ringbuffer;
         }
 
-        RingbufferConfig ringbufferConfig = getRingbufferConfig(name);
         ringbuffer = new RingbufferContainer(
                 name,
-                ringbufferConfig,
+                getRingbufferConfig(name),
                 nodeEngine.getSerializationService(),
                 nodeEngine.getConfigClassLoader());
         ringbuffer.getStore().instrument(nodeEngine);
         containers.put(name, ringbuffer);
         return ringbuffer;
     }
 
-    private RingbufferConfig getRingbufferConfig(String name) {
-        Config config = nodeEngine.getConfig();
-        return config.getRingbufferConfig(getConfigName(name));
-    }
-
     public void addRingbuffer(String name, RingbufferContainer ringbuffer) {
         checkNotNull(name, "name can't be null");
         checkNotNull(ringbuffer, "ringbuffer can't be null");
-        final RingbufferConfig config = nodeEngine.getConfig().getRingbufferConfig(name);
-        final SerializationService serializationService = nodeEngine.getSerializationService();
         ringbuffer.init(
                 name,
-                config,
-                serializationService,
+                getRingbufferConfig(name),
+                nodeEngine.getSerializationService(),
                 nodeEngine.getConfigClassLoader());
         ringbuffer.getStore().instrument(nodeEngine);
         containers.put(name, ringbuffer);
     }
+
+    private RingbufferConfig getRingbufferConfig(String name) {
+        Config config = nodeEngine.getConfig();
+        return config.findRingbufferConfig(getConfigName(name));
```

Comment:
That's true, but all other code uses `find*Config()` to get the read-only config. So the RingBuffer (and actually the Quorum) are "vulnerable" to runtime config changes, without dynamic configs being supported so far.